src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -39,77 +34,55 @@ public override void Initialize()
             var engine = this.Parent as PoAConsensusRuleEngine;
 
             this.slotsManager = engine.SlotsManager;
+            this.federationHistory = engine.FederationHistory;
```

Comment:
As per previous comment, inject the `FederationHistory` directly into the rule's constructor